config/config.hocon.sample

```diff
@@ -26,7 +26,10 @@
       "path": "s3://acme-snowplow-output/raw/",
 
       # Partitioning format; Optional
-      "dateFormat": "%Y-%M-%d",
+      # Valid substitutions are {vendor}, {name}, {format}, {model} for self-describing jsons
+      # and {yy}, {mm}, {dd}, {hh} for year, month, day, hour
+      partitionFormat: "{vendor}.{name}/model={model}/date={yy}-{mm}-{dd}"
```

Comment:
Shouldn't `model` be `version`?

Reply:
Is your question about whether to use model as the discriminator, vs full schema ver (e.g. 1-0-0).  Or is it about whether to call it "version" in the hocon, which is more user-friendly?

I assumed that users would want to put new additions/revisions into the same directory, as they are non-breaking changes for most consumers.  I assumed that only if the first number (model) of the schemaver changes then user would want to put data into a different directory, because it probably breaks most consumers to have incompatible data together.

Reply:
My question is whether `version` is not more user-friendly but your comment made me think that it might actually be confusing, since people are likely to assume it applies to the full `1-x-x` schema ver.

Do we not think though that someone might want to partition more granularly than model? Should `revision` and `patch` also be valid substitutions?

Reply:
I looked to add `revision` and `patch` as template configuration options.  However, it was more difficult than expected because [currently the loader is hard-coded to group by model only](https://github.com/snowplow/snowplow-s3-loader/blob/2.0.0/src/main/scala/com/snowplowanalytics/s3/loader/processing/RowType.scala#L27).

So I think it is a _different_ feature to add grouping by revision/patch, beyond the scope of current issue which is just concerned with templating the s3 directory.

Reply:
I will add to that comment - to make it sound like I'm not just being lazy.  To me it makes complete sense to group by model, and it mirrors how we group data in other scenarios, e.g. when we load to Snowflake or Redshift.  All events with the same model land in the same place.

So I am comfortable with being lazy here, keeping the existing opinionated behaviour, and not supporting grouping by revision/patch unless it is requested as a feature.

Reply:
Agree with everything you said. My only concern is how do we make it clear what `{model}` is. It assumes familiarity with schema ver, which maybe we shouldn't be assuming? Maybe a line in docs somewhere can refer to that and explain what `model` is.